src/net.cpp

```diff
@@ -682,18 +674,27 @@ bool CNode::ReceiveMsgBytes(const char *pch, unsigned int nBytes, bool& complete
         pch += handled;
         nBytes -= handled;
 
-        if (msg.complete()) {
+        if (msg.complete())
+            complete = true;
+    }
+
+    int64_t nTime = GetTimeMicros();
+    LOCK(cs_vRecvMsg);
+    nRecvBytes += nBytes;
```

Comment:
Sure, that just means more locking. Though this is completely uncontested now, so that's not really a problem.